src/main/scala/com/yotpo/metorikku/output/writers/elasticsearch/ElasticsearchOutputWriter.scala

```diff
@@ -0,0 +1,46 @@
+package com.yotpo.metorikku.output.writers.elasticsearch
+
+import com.yotpo.metorikku.configuration.job.output.Elasticsearch
+import com.yotpo.metorikku.output.Writer
+import org.apache.log4j.LogManager
+import org.apache.spark.sql.{DataFrame, SaveMode}
+
+
+class ElasticsearchOutputWriter(props: Map[String, String], elasticsearchOutputConf: Option[Elasticsearch]) extends Writer {
+
+  case class ElasticsearchOutputProperties(saveMode: SaveMode, resource: String, mappingId: String, writeOperation: String,
+                                           port: Int, indexAutoCreate: Boolean, nodesWanOnly: Boolean)
+
+  val log = LogManager.getLogger(this.getClass)
+
+  val saveMode: SaveMode = SaveMode.valueOf(props.getOrElse("saveMode", "Overwrite"))
+  val port: Int = props.getOrElse("port", "9200").toString.toInt
+  val indexAutoCreate: Boolean = props.getOrElse("indexAutoCreate", true).toString.toBoolean
+  val nodesWanOnly: Boolean = props.getOrElse("nodesWanOnly", true).toString.toBoolean
+
+  val dbOptions = ElasticsearchOutputProperties(saveMode,
+                                                props("resource"),
+                                                props.getOrElse("mappingId", ""),
+                                                props.getOrElse("writeOperation", ""),
+                                                port,
+                                                indexAutoCreate,
```

Comment:
Like you did above can't you the same here? don't save to variables before?

Reply:
Your'e right. No Reason.